### PR TITLE
feat: Feature lifecycle sql store

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

* db migration for lifecycle stages
* sql store based on the new migration and tested through the e2e test (if there's error in any query the tests catches it)
* making feature lifecycle service an event emitter itself so that we can wait to stage change events in tests instead of `await ms(randomTime)`

### CODE REVIEW

File: src/lib/features/feature-lifecycle/feature-lifecycle-store.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    async get(feature: string): Promise<FeatureLifecycleView> {
+        const results = await this.db('feature_lifecycles')
+            .where({ feature })
+            .orderBy('created_at', 'asc');
+
+        return results.map(({ stage, created_at }: DBType) => ({
+            stage,
+            enteredStageAt: created_at,
```
Comment (kwasniew) [inline src/lib/features/feature-lifecycle/feature-lifecycle-store.ts:40]: will probably rename enteredStageAt to createdAt in the service and openapi

File: src/lib/features/feature-lifecycle/fake-feature-lifecycle-store.ts [hunk 2]
```diff
@@ -8,9 +8,12 @@ export class FakeFeatureLifecycleStore implements IFeatureLifecycleStore {
     private lifecycles: Record<string, FeatureLifecycleView> = {};
 
     async insert(featureLifecycleStage: FeatureLifecycleStage): Promise<void> {
-        const existing = await this.get(featureLifecycleStage.feature);
+        if (await this.stageExists(featureLifecycleStage)) {
```
Comment (kwasniew) [inline src/lib/features/feature-lifecycle/fake-feature-lifecycle-store.ts:11]: prevent double inserts in fake impl

File: src/migrations/20240405120422-add-feature-lifecycles.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+exports.up = function(db, cb) {
+    db.runSql(
+        `
+        CREATE TABLE IF NOT EXISTS feature_lifecycles (
+            feature VARCHAR(255) NOT NULL REFERENCES features(name) ON DELETE CASCADE,
```
Comment (kwasniew) [inline src/migrations/20240405120422-add-feature-lifecycles.js:7]: we want to delete feature lifecycle when feature is removed

File: src/migrations/20240405120422-add-feature-lifecycles.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+            stage VARCHAR(255) NULL,
+            created_at TIMESTAMP WITH TIME ZONE default now(),
```
Comment (kwasniew) [inline src/migrations/20240405120422-add-feature-lifecycles.js:9]: sticking to generic name instead of specific stage_entered_at

Comment (kwasniew) [reply src/migrations/20240405120422-add-feature-lifecycles.js:9]: using timestamp with timezone to have the same type as features table

**Review by andreas-unleash — APPROVED** [review]
LGTM

**Pull request merged**